sklearn/metrics/cluster/unsupervised.py

```diff
@@ -133,6 +160,21 @@ def silhouette_samples(X, labels, metric='euclidean', **kwds):
         allowed by :func:`sklearn.metrics.pairwise.pairwise_distances`. If X is
         the distance array itself, use "precomputed" as the metric.
 
+    blockwise: {'auto', True, False}
+        Enables blockwise computation of the distance matrix.
+        If false, the full distance matrix is computed yielding in fast
+        computation but high memory consumption. If True, it computes
+        clusterwise distance matrices, dividing memory consumption by
+        approximately the squared number of clusters. The latter allows
+        parallelization through ``n_jobs`` parameter.
+        Default is 'auto' that chooses the fastest option without
+        consideration for memory consumption.
```

Comment:
here just after this line add a ``.. versionadded::0.17`` not sure if it needs a newline before it though